internal/controller/nlb/flowlog/flowlog.go

```diff
@@ -0,0 +1,215 @@
+type externalFlowLog struct {
+	// A 'client' used to connect to the externalFlowLog resource API. In practice this
+	// would be something like an IONOS Cloud SDK client.
+	service              flowlog.NLBFlowLog
+	log                  logging.Logger
+	isUniqueNamesEnabled bool
+}
+
+func (c *externalFlowLog) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.FlowLog)
+	if !ok {
+		return managed.ExternalObservation{}, errNotFlowLog
+	}
+	flowLogID := meta.GetExternalName(cr)
+	if flowLogID == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	datacenterID := cr.Spec.ForProvider.DatacenterCfg.DatacenterID
+	nlbID := cr.Spec.ForProvider.NLBCfg.NetworkLoadBalancerID
+	observed, err := c.service.GetFlowLogByID(ctx, datacenterID, nlbID, flowLogID)
+	if err != nil {
+		if errors.Is(err, flowlog.ErrNotFound) {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, err
+	}
+	flowlog.SetStatus(cr, observed)
+	cr.Status.AtProvider.FlowLogID = flowLogID
+	c.log.Debug(fmt.Sprintf("Observing state: %v", cr.Status.AtProvider.State))
+	clients.UpdateCondition(cr, cr.Status.AtProvider.State)
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  flowlog.IsUpToDate(cr, observed),
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+func (c *externalFlowLog) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) { // nolint: gocyclo
+	cr, ok := mg.(*v1alpha1.FlowLog)
+	if !ok {
+		return managed.ExternalCreation{}, errNotFlowLog
+	}
+	cr.SetConditions(xpv1.Creating())
+	// Check external name in order to avoid duplicates,
+	// since the creation requests take longer than other resources.
+	if meta.GetExternalName(cr) != "" {
+		return managed.ExternalCreation{}, nil
+	}
+	if cr.Status.AtProvider.State == compute.BUSY {
+		return managed.ExternalCreation{}, nil
+	}
+
+	datacenterID := cr.Spec.ForProvider.DatacenterCfg.DatacenterID
+	nlbID := cr.Spec.ForProvider.NLBCfg.NetworkLoadBalancerID
+	if c.isUniqueNamesEnabled {
+		// isUniqueNamesEnabled option enforces FlowLog names to be unique per Datacenter and NetworkLoadBalancer
+		// Multiple Flow Logs with the same name will trigger an error
+		// If only one instance is found, it will be "imported"
+		flowLogDuplicateID, err := c.service.CheckDuplicateFlowLog(ctx, datacenterID, nlbID, cr.Spec.ForProvider.Name)
+		if err != nil {
+			return managed.ExternalCreation{}, err
+		}
+		if flowLogDuplicateID != "" {
+			cr.Status.AtProvider.FlowLogID = flowLogDuplicateID
+			meta.SetExternalName(cr, flowLogDuplicateID)
+			return managed.ExternalCreation{}, nil
+		}
+	}
+	flowLogInput := flowlog.GenerateCreateInput(cr)
+	newInstance, err := c.service.CreateFlowLog(ctx, datacenterID, nlbID, flowLogInput)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	cr.Status.AtProvider.FlowLogID = *newInstance.Id
```

Comment:
this won't be stored.

Reply:
will change, `cr.Status` shouldn't be set by `Create` anyway